src/client/ui/cursor/index.js

```diff
@@ -1,97 +1,141 @@
+import hammerhead from './../deps/hammerhead';
 import testCafeCore from './../deps/testcafe-core';
-import CursorBehavior from './behavior';
-import CursorIFrameBehavior from './iframe-behavior';
+import CURSOR_UI_MESSAGES from './messages';
+
+
+var Promise        = hammerhead.Promise;
+var shadowUI       = hammerhead.shadowUI;
+var browserUtils   = hammerhead.utils.browser;
+var messageSandbox = hammerhead.eventSandbox.message;
+
+var styleUtils    = testCafeCore.styleUtils;
+var positionUtils = testCafeCore.positionUtils;
+var eventUtils    = testCafeCore.eventUtils;
+
+const CURSOR_CLASS       = 'cursor';
+const TOUCH_CLASS        = 'touch';
+const L_MOUSE_DOWN_CLASS = 'l-mouse-down';
+const R_MOUSE_DOWN_CLASS = 'r-mouse-down';
+const STATE_CLASSES      = [L_MOUSE_DOWN_CLASS, R_MOUSE_DOWN_CLASS].join(' ');
+
+// Setup cross-iframe interaction
+messageSandbox.on(messageSandbox.SERVICE_MSG_RECEIVED_EVENT, e => {
+    var msg = e.message;
+
+    switch (msg.cmd) {
+        case CURSOR_UI_MESSAGES.moveRequest:
+            var position = positionUtils.getIframePointRelativeToParentFrame({ x: msg.x, y: msg.y }, e.source);
+
+            CursorUI
+                .move(position.x, position.y)
+                .then(() => messageSandbox.sendServiceMsg({ cmd: CURSOR_UI_MESSAGES.moveResponse }, e.source));
+            break;
+
+        case CURSOR_UI_MESSAGES.leftButtonDownRequest:
+            CursorUI
+                .leftButtonDown()
+                .then(() => messageSandbox.sendServiceMsg({ cmd: CURSOR_UI_MESSAGES.leftButtonDownResponse }, e.source));
+            break;
+        case CURSOR_UI_MESSAGES.rightButtonDownRequest:
+            CursorUI
+                .rightButtonDown()
+                .then(() => messageSandbox.sendServiceMsg({ cmd: CURSOR_UI_MESSAGES.rightButtonDownResponse }, e.source));
+            break;
+        case CURSOR_UI_MESSAGES.buttonUpRequest:
+            CursorUI
+                .buttonUp()
+                .then(() => messageSandbox.sendServiceMsg({ cmd: CURSOR_UI_MESSAGES.buttonUpResponse }, e.source));
+            break;
+    }
+});
+
+var CursorUI = {
+    cursorElement:  null,
+    x:              50,
+    y:              50,
+    pointerOffsetX: 0,
+    pointerOffsetY: 0,
+
+    _createElement () {
+        this.cursorElement = document.createElement('div');
+        shadowUI.addClass(this.cursorElement, CURSOR_CLASS);
+
+        // NOTE: For IE in Cross domain iframe we can't use touch cursor
+        // because we won't be able to get element under cursor
```

Comment:
```
        // NOTE: For IE, we can't use the touch cursor in a cross-domain iframe
        // because we won't be able to get an element under the cursor
```